### DISCUSSION TITLE
setting up ImGUI with GLUT

### DISCUSSION THREAD

**Discussion by cozerCode** [ocornut/imgui#4745] [New Users Build/Link/Run issues ONLY!]
> <h3>hi,</h3>
> I'm new to GUI and I decided to use ImGUI with GLUT, but I only found GLFW tutorials, so can you help me to do this? <be>
> <h4>Thanks.</h4>

edit i may found a way to set it up

**Comment by ocornut** [comment]
There’s an example app doing exactly that in the examples/ folder…

**Reply by cozerCode** [reply]
fixed by adding ```-c C:/GL/IMGUI/*.cpp  -c C:/GL/IMGUI/backends/imgui_impl_glut.cpp``` to the build command, you can close this. Thanks!

**Comment by cozerCode** [comment]
thanks for the reply i managed to do some progress and i'm getting;
````dir/main.cpp:119: undefined reference to `ImGui::ShowDemoWindow(bool*)'
D:\dir/main.cpp:126: undefined reference to `ImGui::Begin(char const*, bool*, int)'
D:\dir/main.cpp:128: undefined reference to `ImGui::Text(char const*, ...)'
D:\dir/main.cpp:129: undefined reference to `ImGui::Checkbox(char const*, bool*)'
D:\dir/main.cpp:130: undefined reference to `ImGui::Checkbox(char const*, bool*)'
D:\dir/main.cpp:132: undefined reference to `ImGui::SliderFloat(char const*, float*, float, float, char const*, int)'
D:\dir/main.cpp:133: undefined reference to `ImGui::ColorEdit3(char const*, float*, int)'
D:\dir/main.cpp:135: undefined reference to `ImGui::Button(char const*, ImVec2 const&)'
D:\dir/main.cpp:137: undefined reference to `ImGui::SameLine(float, float)'````
there is more errors than that but this is some
i think i need to include or link something?
here is my build cmd:
````g++ -c src/*.cpp -std=c++17 -g -Wall -m32 -I include -I C:/GL/GLUT/include/GL -I C:/GL/IMGUI -I C:/GL/IMGUI/backends && g++ *.o -o bin/debug/main -L C:/GL/GLUT/lib/ -lmingw32 -lopengl32 -lglu32 -lfreeglut && start bin/debug/main````
i have the same exact code in examples\example_glut_opengl2zmain.cpp

btw i use gcc compiler
